config.c

```diff
@@ -1437,6 +1437,14 @@ int git_config_bool(const char *name, const char *value)
 	return v;
 }
 
+int git_config_bool_or_default(const char *name, const char *value, const char default_value)
```

Comment:
As pointed out by the many failures in the CI run (e.g. https://github.com/git/git/runs/7763398620?check_suite_focus=true#step:4:161), this would need a prototype declaration in `config.h`.

And since it is a new function that is broader in scope than just the `rebase.autostash` setting, it needs to be split out into its own commit.

Having said that, have you considered simply changing the caller in `builtin/pull.c` from `git_config_bool()` to `git_config_get_maybe_bool()`?